src/bespokelabs/curator/llm/llm.py

```
@@ -234,48 +235,68 @@ def __call__(
+            prompt_func_source = _get_function_source(self.prompt_formatter.prompt_func)
+            if self.prompt_formatter.parse_func is not None:
+                parse_func_source = _get_function_source(self.prompt_formatter.parse_func)
+            else:
+                parse_func_source = ""
+
+            metadata_dict = {
+                "timestamp": datetime.now().isoformat(),
+                "dataset_hash": dataset_hash,
+                "prompt_func": prompt_func_source,
+                "parse_func": parse_func_source,
+                "model_name": self.prompt_formatter.model_name,
+                "response_format": (
+                    str(self.prompt_formatter.response_format.model_json_schema())
+                    if self.prompt_formatter.response_format
+                    else "text"
+                ),
+                "run_hash": fingerprint,
+                "batch_mode": self.batch_mode,
+            }
+            metadata_db.store_metadata(metadata_dict)
+
+        import tempfile
+
+        # Check if cache is disabled
+        cache_disabled = os.getenv("CURATOR_DISABLE_CACHE", "").lower() in ["true", "1"]
+        if cache_disabled:
+            logger.info("Cache disabled by CURATOR_DISABLE_CACHE env variable.")
+            # Create a temporary directory that will be automatically cleaned up
+            run_cache_dir = tempfile.mkdtemp()
+            logger.debug(f"Created temporary directory for non-cached run: {run_cache_dir}")
+            # Skip metadata storage when cache is disabled
+            metadata_db = None
         else:
-            dataset = self._request_processor.run(
-                dataset=dataset,
-                working_dir=run_cache_dir,
-                parse_func_hash=parse_func_hash,
-                prompt_formatter=self.prompt_formatter,
-            )
+            run_cache_dir = os.path.join(curator_cache_dir, fingerprint)
+
+        try:
+            if batch_cancel:
```

batch_cancel and cache_disabled should never happen at the same time. we should throw an error if both are true.